ui/src/logger/logger_debug.rs

```diff
@@ -0,0 +1,89 @@
+use std::time::Duration;
+
+use chrono::Local;
+use dioxus::prelude::*;
+
+use dioxus_desktop::use_window;
+use kit::elements::label::Label;
+use tokio::time::sleep;
+
+use crate::components::settings::sub_pages::developer::WindowDropHandler;
+
+use super::logger::{Logger, LOGGER};
+
+const STYLE: &str = include_str!("./style.scss");
+
+#[inline_props]
+#[allow(non_snake_case)]
+pub fn LoggerDebug(cx: Scope, _drop_handler: WindowDropHandler) -> Element {
+    Logger::activate_logger();
+    let window = use_window(cx);
+
+    let logger = LOGGER.read();
+    let logs = logger.show_log();
+
+    let logs_to_show = use_state(cx, || logs.clone());
+
+    let logs_on_screen_len = use_ref(cx, || 0);
+
+    let now = Local::now();
+    let formatted_datetime = now.format("%a %b %d %H:%M:%S").to_string();
+    let debug_logger_started_time = use_ref(cx, || formatted_datetime.clone());
+
+    let script = include_str!("./script.js");
+
+    use_future(cx, (), |_| {
+        to_owned![logs_to_show, window, script, logs_on_screen_len];
+        async move {
+            loop {
+                sleep(Duration::from_millis(100)).await;
+                let new_logs = logger.show_log();
```

Comment:
Since this is being used in a `use_future`, we should 

A. Spawn a blocking task since the operations within `Logger::show_log` are blocking
B. Change the logic of the `Logger` to be async. 

I think A would be better. You would use `tokio::spawn_blocking` with the logger passed into it. That way, its reading operations would not block other operating task in the same thread. Of course, B would be preferred (since its a easy switch) but since its also being used outside of the future, we cannot really do that unless you want to spawn a use_future for the code above in the file, but that may not be desirable. 


```suggestion
                let new_logs = tokio::task::spawn_blocking(move || LOGGER.read().show_log()).await.expect("Should not panic");
```

Note: This should not panic unless there is an issue with the tokio runtime in which case might be a bug if the runtime is still operating. A safer option would be to do something like

```rust
let Ok(new_logs) = tokio::task::spawn_blocking(move || LOGGER.read().show_log()).await else { break };
```

Reply:
Darius, I changed code, to read log entries now, and not more read file every time, just in the beginning, when open debug logger.